calcom/cal.com
# fix: embed hides floating button

## What does this PR do?

- Increased modalbox z-index to highest. It would stay on top of button because it is added after the button.
- Also, some code cleanup that was required for hiding floating button popup which isn't required now.

-- Edit by hariombalhara

Fixes #8851

https://github.com/calcom/cal.com/assets/56112399/02362901-8a5b-478b-9278-b08323c49549

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?



- ran the embed tests from the root directory - 4 errors, 5 skips, 1 successful (same as that of in production)
- in terms of manual testing, ensured that the functionality of the floating popup is alright by going through the entire flow (click -> change timezone -> search timezone -> select date -> change time format -> select time -> update details -> confirm)
- ensured that the button is hidden when the Modal is opened and is shown back again when the modal is closed (in any possible way: close button, click away, cancel button)

## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
